teslajsonpy/controller.py

```diff
@@ -1540,9 +1541,36 @@ def update_interval(self) -> int:
     @update_interval.setter
     def update_interval(self, value: int) -> None:
         """Set update_interval."""
+        if value < 0:
+            value = 300
```

Comment:
Sorry, didn't review line by line until now. Create a constant for this and then use it in your test like you have below. We shouldn't have hard coded default numbers buried in the code.

After this, I think it should be good to merge.

Reply:
Yeah, I saw it as hard coded hence kept it that way. Changed it to a constant instead. Updated tests as well to test that by default it is set to this constant and for checking.

Reply:
Ooops. I guess we did that a while back if that came from us. Thanks for fixing it. Looks good. I'll merge this an do a release.

Reply:
> Ooops. I guess we did that a while back if that came from us. Thanks for fixing it. Looks good. I'll merge this an do a release.

The value = 300 you marked came from me. Did it that way as there was no constant defined for the default in the__init__.
That's what I meant with not hard-coded, did not see it as a constant for the __init__ declaration. :-)